subsys/greybus/platform/service.c

```diff
@@ -40,21 +40,20 @@ gb_transport_get_backend(void)
 	return xport;
 }
 
-int greybus_service_init(const struct device *bus)
+static int greybus_service_init(const struct device *bus)
 {
-    int r;
+	int r = -EINVAL;
 	uint8_t *mnfb;
 	size_t mnfb_size;
-    unsigned int *cports = NULL;
+	unsigned int *cports = NULL;
 
-    LOG_DBG("Greybus initializing..");
-
-	r = gb_service_deferred_init();
-	if (r < 0) {
-		LOG_ERR("gb_service_deferred_init() failed: %d", r);
-		goto out;
+	if (xport != NULL) {
+		LOG_ERR("service already initialized");
+		return -EALREADY;
 	}
 
+	LOG_DBG("Greybus initializing..");
+
 	bus = device_get_binding(GREYBUS_BUS_NAME);
```

Comment:
I think the warning (promoted to error) is actually here. If `bus` is `NULL`, then set `r = -ENODEV;` before going to `out`.